### PR TITLE
new(userspace/libsinsp): expose `get_owner_last_error` in plugin's capture listening capability

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(userspace/libsinsp): expose `get_owner_last_error` in plugin's capture listening capability
```

### CODE REVIEW

File: userspace/plugin/plugin_api.h [hunk 1]
```diff
@@ -469,6 +469,11 @@ typedef struct ss_plugin_capture_listen_input {
 	//
 	// Vtable for controlling a state table for write operations.
 	ss_plugin_table_writer_vtable_ext* table_writer_ext;
+	//
+	// Return a string with the error that was last generated by the plugin's
+	// owner, or NULL if no error is present.
+	// The string pointer is owned by the plugin's owenr.
+	const char* (*get_owner_last_error)(ss_plugin_owner_t* o);
```
Comment (FedeDP) [inline userspace/plugin/plugin_api.h:476]: This is needed to expose `table_reader` and `table_writer` from plugin sdks.

File: userspace/libsinsp/sinsp.cpp [hunk 2]
```diff
@@ -371,7 +371,10 @@ void sinsp::open_common(scap_open_args* oargs,
 	// notify registered plugins of capture open
 	for(auto& p : m_plugin_manager->plugins()) {
 		if(p->caps() & CAP_CAPTURE_LISTENING) {
-			p->capture_open();
+			if(!p->capture_open()) {
```
Comment (FedeDP) [inline userspace/libsinsp/sinsp.cpp:374]: Manage errors in capture_open/close.

**Review by LucaGuerra — APPROVED** [review]

**Review by jasondellaluce — APPROVED** [review]

**Review by leogr — APPROVED** [review]

**Pull request merged**